### PR TITLE
[PF-599] Make controlled bucket creation synchronous

### PR DESCRIPTION
**Pull request by zloery**
Currently, controlled GCS bucket creation is asynchronous. However, the call completes very quickly and this can be made synchronous. While this pattern works for GCS buckets, I don't know that all controlled resource creation can be synchronous. To support this, I've left the asynchronous creation method in the service and moved JobReport out of the common + required request fields of the API object.

Controlled GCS bucket deletion still needs to be asynchronous, as that call can potentially take much longer

### CODE REVIEW

**Comment by zloery** [comment]
Client tests pass with a locally published client, I'll publish the new version once this is approved

File: workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java [hunk 1]
```diff
@@ -258,23 +254,15 @@ private CreatedControlledGcpGcsBucket createBucketAttempt(ControlledGcpResourceA
...
-    logger.info(
-        "Attempt to creating bucket {} jobId {} workspace {}", bucketName, jobId, getWorkspaceId());
-    CreatedControlledGcpGcsBucket bucket = resourceApi.createBucket(body, getWorkspaceId());
-    while (ClientTestUtils.jobIsRunning(bucket.getJobReport())) {
-      TimeUnit.SECONDS.sleep(CREATE_BUCKET_POLL_SECONDS);
-      bucket = resourceApi.getCreateBucketResult(getWorkspaceId(), jobId);
-    }
-    logger.info("Create bucket status is {}", bucket.getJobReport().getStatus().toString());
-    return bucket;
+    logger.info("Attempt to creating bucket {} workspace {}", bucketName, getWorkspaceId());
```
Comment (jaycarlton) [inline workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java:264]: *create

**Review by jaycarlton — APPROVED** [review]

File: src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java [hunk 2]
```diff
@@ -43,6 +44,29 @@ public ControlledResourceService(
     this.stageService = stageService;
   }
 
+  public ControlledResource syncCreateControlledResource(
```
Comment (ddietterich) [inline src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java:47]: Ah, good idea to leave the async code. We are likely to need it.

File: workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java (hunk 1 above)
Comment (ddietterich) [reply workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java:264]: My typo, but thanks for fixing it :)

**Review by ddietterich — APPROVED** [review]
LGTM

**Comment by zloery** [comment]
published `0.21.0-SNAPSHOT` from this branch

**Pull request merged**